### DISCUSSION TITLE
🔨 Breaking changes in upcoming SysReptor release

### DISCUSSION THREAD

**Discussion by aronmolnar** [Syslifters/sysreptor#230] [Announcements]
# TLDR
We will release a new SysReptor version with breaking changes on April 10.  
You might need to take actions before you upgrade.

**Web servers**
* SysReptor Cloud users: No action needed
* Caddy: No action needed
* nginx: Make sure your nginx configuration allows websockets (see https://github.com/Syslifters/sysreptor/blob/main/deploy/sysreptor.nginx#L27 and https://github.com/Syslifters/sysreptor/blob/main/deploy/sysreptor.nginx#L36)
* Other web servers: Make sure your web server supports websockets

**reptor (CLI tool)**
* Update reptor to the latest version (`pip3 install --upgrade reptor`)

# Breaking changes in SysReptor

On April 10, we will introduce collaborative editing for notes.

![image](https://github.com/Syslifters/sysreptor/assets/25219407/d9e7c3e1-d656-4f44-8f2a-be80b2b94a34)

We are implementing this feature using websockets. Notes are then no longer locked while users are editing them.  
This introduces breaking changes in some **web server configurations** and our **CLI tool `reptor`**.

## Web servers
If your web server doesn't support web sockets, some SysReptor functionalities might not work as expected in the future.  

### Deployments with Caddy web server (our recommended solution)
If you deployed Caddy [following our documentation](https://docs.sysreptor.com/setup/webserver/#caddy-recommended), there is no action needed.  
Caddy supports websockets out of the box.

### Deployments with nginx web server
We updated our nginx [sample configuration](https://github.com/Syslifters/sysreptor/blob/main/deploy/sysreptor.nginx) to support websockets. Make sure that your nginx configuration is updated accordingly.

### Deployments with other web servers
Make sure your web server supports websockets.

## reptor
Update reptor to the latest version (`pip3 install --upgrade reptor`).  
The changes are backwards-compatible.
